client/src/pages/lessons/edit.tsx

```diff
@@ -159,6 +169,26 @@ const LessonEdit = (props: {
     return /^[a-z0-9-]+$/g.test(lesson.lessonId);
   }
 
+  function isExpValid(exp: LessonExpectation): boolean {
+    if (!exp.features) {
+      return true;
+    }
+    const ajv = new Ajv({ allErrors: true, verbose: true });
```

Comment:
looks to me lik the Ajv validator is stateless/functional, so for that and DRY-principal, let's move validation out into a `validation.ts` module, and then in there we can make sure schema gets read and compiled only one time per app load, e.g.

```typescript
// validation.ts

// the validator should be private to the module
// and global (only created once at most
let _expectationFeaturesValidator: Ajv.ValidateFunction;

function getExpectationFeaturesValidator() : Ajv.ValidateFunction {
  if(!_expectationFeaturesValidator) {
    _expectationFeaturesValidator = new Ajv({ allErrors: true, verbose: true }).compile(require("schemas/expectation-feature-schema.json"));
  }
  return _expectationFeaturesValidator;
}

export function validateExpectationFeatures(json: any): boolean {
  return getExpectationFeaturesValidator()(json);
}